services/api/src/service/administratorOrganization.ts

```diff
@@ -0,0 +1,181 @@
+import {
+    organizationMetadataTable,
+    userOrganizationMappingTable,
+} from "@/schema.js";
+import type { PostgresJsDatabase } from "drizzle-orm/postgres-js";
+import { and, eq } from "drizzle-orm";
+import { useCommonUser } from "./common.js";
+import { httpErrors } from "@fastify/sensible";
+import { log } from "@/app.js";
+
+interface RemoveUserOrganizationMappingProps {
+    db: PostgresJsDatabase;
+    username: string;
+    organizationName: string;
+}
+
+export async function removeUserOrganizationMapping({
+    db,
+    username,
+    organizationName,
+}: RemoveUserOrganizationMappingProps) {
+    const { getUserIdFromUsername } = useCommonUser();
+    const targetUserId = await getUserIdFromUsername({
+        db: db,
+        username: username,
+    });
+
+    try {
+        const organizationId = await getOrganizationIdFromOrganizationName(
+            db,
+            organizationName,
+        );
+        if (organizationId == undefined) {
+            throw httpErrors.notFound("Failed to locate organization ID");
+        } else {
+            const deletedMapping = await db
+                .delete(userOrganizationMappingTable)
+                .where(
+                    and(
+                        eq(userOrganizationMappingTable.userId, targetUserId),
+                        eq(
+                            userOrganizationMappingTable.organizationMetadataId,
+                            organizationId,
+                        ),
+                    ),
+                )
+                .returning();
+            if (deletedMapping.length == 0) {
+                throw httpErrors.internalServerError(
+                    "Organization mapping does not exist",
+                );
+            }
+        }
+    } catch (err: unknown) {
+        log.error(err);
+        throw httpErrors.internalServerError(
```

Comment:
you can remove this try-catch, the stacktrace should log correctly with a 500 status, when the fastify global handler catches it